plugins/foreman_salt/13.0/index.md

```diff
@@ -0,0 +1,444 @@
+
+## 4.7 Grains
+
+Cached grains are uploaded to Foreman when an external node classifier is run (e.g. when running state.highstate).  You can view them in Foreman's Fact browser.  Structured facts are navigable by clicking on the name and drilling down in child values.
+
+![](/static/images/plugins/foreman_salt/grains.png)
+
+## 4.8 Reporting
+
+When running `state.highstate`, you can have Foreman process the results and show them in the UI. These reports show statistics about what was done, how long it took, complete with pie charts and file diffs.
+
+![](/static/images/plugins/foreman_salt/reports.gif)
+
+Uploading the salt reports is done by `/usr/sbin/upload-salt-reports` and is scheduled by a cron job running on the smart proxy.  By default, reports are uploaded to Foreman once every 10 minutes from the Salt master's job cache. You may modify the smart_proxy_salt cron job to customize this by editing `/etc/cron.d/smart_proxy_salt`.
+
+<b>Why not use a returner?</b>
+
+Support for a [custom returner](http://docs.saltstack.com/en/latest/ref/returners/) would make this cron on the master unnecessary, however, there are limitations on both the Salt and Foreman Proxy side that make this impossible to do securely (for now).
+
+### 4.8.1 Scheduling
+
+You can schedule your salt jobs like it is described in [Salt Job Management](https://docs.saltstack.com/en/latest/topics/jobs/). E.g. add the following to the /etc/salt/minion configuration file on your minions:
+
+    schedule:
+      highstate:
+        function: state.highstate
+        minutes: 5
+
+If foreman_remote_execution is installed, you can add a Scheduled / Recurring Job to run Salt `state.highstate` or any other Salt function.
+
+## 4.9 API
+
+Foreman Salt extends v2 of the Foreman RESTful API with Salt-specific features.  See the Foreman manual for general info on the API, and view the full API documentation is available on your Foreman server: `http://foreman.example.com/apidoc/v2.html`.
+
+Using Curl to get a list of keys from "smartproxy.example.com":
+
+    curl -k -u admin:changeme -H "Accept: version=2,application/json" https://localhost/salt/api/v2/salt_keys/smartproxy.example.com
+
+## 4.10 CLI
+
+A [hammer](/manuals/latest/index.html#4.5CommandLineInterface) plugin for Salt is available, [follow the instructions for installing a plugin](/plugins/#2.Installation) to install hammer_cli_foreman_salt, and then see `hammer --help` for more information.
+
+Examples:
+
+  - Creating a Salt State:
+
+<pre>
+hammer salt-state create --name foo.bar
+</pre>
+
+  - Viewing info about a minion:
+
+<pre>
+hammer salt-minion info --name minion.example.com
+</pre>
+
+  - Adding states to a minion:
+
+<pre>
+hammer salt-minion update --name minion.example.com --salt-states foo.bar
+</pre>
+
+
+# 5 Help
+
+## 5.1 Troubleshooting
+
+If you find a bug, please file it in [Redmine](http://projects.theforeman.org/projects/salt/issues/new).  You can find us on freenode on #theforeman as well.
```

Comment:
I'm always a fan of linking to https when available to avoid the http redirect
```suggestion
If you find a bug, please file it in [Redmine](https://projects.theforeman.org/projects/salt/issues/new).  You can find us on freenode on #theforeman as well.
```